src/vdaf/hits.rs

```diff
@@ -522,24 +538,37 @@ impl<I: Idpf<2, 2>> Aggregator for Hits<I> {
     ) -> Result<Vec<I::Field>, VdafError> {
         let mut agg_share = vec![I::Field::zero(); agg_param.len()];
         for output_share in output_shares.into_iter() {
-            if output_share.len() != agg_share.len() {
-                return Err(VdafError::Uncategorized(format!(
-                    "unexpected output share length: got {}; want {}",
-                    output_share.len(),
-                    agg_share.len()
-                )));
-            }
-
-            for (x, y) in agg_share.iter_mut().zip(output_share) {
-                *x += y;
-            }
+            self.accumulate(agg_param, &mut agg_share, &output_share)?;
         }
 
         Ok(agg_share)
     }
+
+    fn accumulate(
+        &self,
+        agg_param: &Self::AggregationParam,
+        accumulator: &mut Self::OutputShare,
+        output_share: &Self::OutputShare,
+    ) -> Result<(), VdafError> {
+        if output_share.len() != agg_param.len() || accumulator.len() != agg_param.len() {
+            return Err(VdafError::Uncategorized(format!(
+                "unexpected accumulator or output share length: got {}/{}; want {}",
+                accumulator.len(),
+                output_share.len(),
+                agg_param.len()
+            )));
+        }
```

Comment:
This error message is a bit confusing. I'd suggest validating the length of `accumulator` and `output_share` on separate lines.